File: php/create_escrow.php
<?php
/*
 * generate salt value
 */
$chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789`~!@#$%^&*()-=_+';
$l = strlen($chars) - 1;
$salt = '';
for ($i = 0; $i < 8; ++$i) {
    $salt .= $chars[rand(0, $l)];
}
$method= "POST";
$APIEndpoint = "/v1/escrow";
$timestamp = time();
$apiKey = <your_api_key>; // string
$apiSecret = <your_api_secret_key> // string
/*
 * generate to_sign
 * to_sign = toUpperCase(Method) + Api-Endpoint + Salt + Timestamp + API-Key + API-Secret
 */
$to_sign = $method.$APIEndpoint.$salt.$timestamp.$apiKey.$apiSecret;
/*
 * generate signature
 * $hmacSHA256 is generate hmacSHA256
 * $signature is convert hmacSHA256 into base64 encode
 * in document: signature = Base64(hmacSHA256(to_sign, API-Secret))
 */
$hmacSHA256 = hash_hmac('sha256', $to_sign, $apiSecret);
$signature = base64_encode($hmacSHA256);
$curl = curl_init();
curl_setopt_array(
    $curl,
    [
      CURLOPT_URL => 'https://api-sandbox.tazapay.com/v1/escrow',
      CURLOPT_RETURNTRANSFER => true,
      CURLOPT_ENCODING => '',
      CURLOPT_MAXREDIRS => 10,
      CURLOPT_TIMEOUT => 0,
      CURLOPT_FOLLOWLOCATION => true,
      CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
      CURLINFO_HEADER_OUT => true,
      CURLOPT_SSL_VERIFYPEER => true,
      CURLOPT_CUSTOMREQUEST => 'POST',
      CURLOPT_POSTFIELDS =>'{
        "txn_type": "goods",
        "release_mechanism": "tazapay",
        "reference_id": "",
        "initiated_by": "a42bd4be-811e-4ff7-8e16-1a052734204c",
        "buyer_id": "",
        "buyer": {
          "country": "SG",
          "email": "send@email.com",
          "ind_bus_type" : "Individual",
          "name": "buyer company/contact-person name"
        },
        "seller_id": "",
        "seller": {
          "country": "IN",
          "email": "send@email.com",
          "ind_bus_type" : "Individual",
          "name": "seller company/contact-person name"
        },
        "txn_description": "txn description",
        "txn_docs": [{
            "type": "Proforma Invoice",
            "url": "https://direct.download.link"
          },
          {
            "type": "Others",
            "name": "OTH - File",
            "url": "https://direct.download.link"
          }
        ],
        "is_milestone": false,
        "attributes": {},
        "invoice_currency": "USD",
        "invoice_amount": 100,
        "fee_tier": "standard",
        "fee_paid_by": "",
        "fee_percentage": 100,
        "release_docs": [{
            "type": "Proof of Work",
            "url": "https://direct.download.link"
          },
          {
            "type": "Others",
            "name": "OTH - File",
            "url": "https://direct.download.link"
          }
        ]
      }',
      CURLOPT_HTTPHEADER => [
        'accesskey: '.$apiKey,
        'salt: '.$salt,
        'signature: '.$signature,
        'timestamp: '.$timestamp,
        'Content-Type: application/json'
      ],
    ]
);
$response = curl_exec($curl);
$info = curl_getinfo($curl);
$header_info = curl_getinfo($curl, CURLINFO_HEADER_OUT);
echo "Response:".$response;
curl_close($curl);

File: java/create_escrow.java
import retrofit2.http.Body;
import retrofit2.http.POST;
import io.qameta.allure.Step;
import retrofit2.Call;
import retrofit2.Response;
import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.*;

public class createEscrow {
    public static void main (String[] args) {
        @POST("/v1/escrow")
        Call<CreateEscrowResponse> createEscrow(@Body CreateEscrowRequest request);
            try {
                CreateEscrowRequest escrowRequest = CreateEscrowRequest
                .builder()
                .txnType("goods")
                .releaseMechanism("tazapay")
                .initiatedBy("bcabf0d5-59a7-490f-bfc4-8b8b9cc47790")
                .buyerId("bcabf0d5-59a7-490f-bfc4-8b8b9cc47790")
                .sellerId("de7692f6-033e-4595-8532-636b3ae15222")
                .txnDescription("Positive transaction")
                .invoiceCurrency("SGD")
                .invoiceAmount(1000)
                .feeTier("standard")
                .feePaidBy("buyer")
                .feePercentage(100)
                .build();
        Call<CreateEscrowResponse> escrow = escrowService.createEscrow(escrowRequest);
        Response<CreateEscrowResponse> response = escrow.execute();
        
    } catch(Exception e) {
        e.printStackTrace;
    }
}
@Data
@Builder
@JsonNaming(value= PropertyNamingStrategy.SnakeCaseStrategy.class)
public class CreateEscrowRequest {
    private String txnType;
    private String releaseMechanism;
    private String buyerId;
    private String sellerId;
    private String txnDescription;
    private String invoiceCurrency;
    private double invoiceAmount;
    private String feeTier;
    private String feePaidBy;
    private float feePercentage;
    private String initiatedBy;
}
@Data
@NoArgsConstructor
@Getter@Setter
public class CreateEscrowResponse {
    private Data data;
    private String status;
    private String message;

@JsonNaming(value = PropertyNamingStrategy.SnakeCaseStrategy.class)
@NoArgsConstructor
@Getter@Setter
    public class Data{
        private String accountId;
        private String customerId;
    }

}

}